helm/chaos-mesh/templates/controller-manager-deployment.yaml

```diff
@@ -47,6 +47,14 @@ spec:
             valueFrom:
               fieldRef:
                 fieldPath: metadata.namespace
+          - name: TEMPLATE_NAMESPACE
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.namespace
+          - name: TARGET_NAMESPACE
+            value: {{ .Values.controllerManager.targetNamespace }}
+          - name: CLUSTER_SCOPED
+            value: "{{ .Values.clusterScoped }}"
```

Comment:
Most controllers I knew is using `starting flag` to indicate the `scope`, Could you keep the same style?

Reply:
Do you mean a command line flags like `--scoped=xxx`?

Or could you give me an example?

Reply:
Exactly, using command line flags is what I mean.

Reply:
Currently, all configuration things in chaos-controller is stored by environment variables. I just want to keep same style.

I agree with your suggestion, cli flags is more friendly to help information and auto-completion. But we still have a lot work to do for transferring from ENV to FLAG.

How about creating another issue to track this?

Reply:
> I agree with your suggestion, cli flags is more friendly to help information and auto-completion. But we still have a lot work to do for transferring from ENV to FLAG.
> 
> How about creating another issue to track this?

https://github.com/chaos-mesh/chaos-mesh/pull/613 https://github.com/chaos-mesh/chaos-mesh/pull/598
@STRRL 
Please don't do it unless it is absolutely necessary.
We've already talked about this in the WeChat group, and we've decided to use env, and keep the same style.

Reply:
@Gallardot @Yisaer 
Aha, actually I don't care using ENV or FLAG. I think the best practice is a configuration layer, stay on the top of the dependency tree. 

Yes, it is `config.ChaosControllerConfig`(and related things). So whatever what we want, we could just modify this pkg. (Maybe we just using viper someday, to build a ~~complex~~  efficient configuration layer.

BTW, a nested `WatcherConfig *watcher.Config` is a really bad idea. It is the inner state of `K8sConfigMapWather`, we should not put it into config struct rudely.

Reply:
So, keep the ENV. 😀